### PR TITLE
Virtio snd over ivshmem

### PR DESCRIPTION
**Pull request by YuanyuanZhao609**
virtio_snd: virtio sound support virtio_shmem
    
    Tracked-On: OAM-124478
    Signed-off-by: Yuanyuan Zhao <yuanyuan.zhao@intel.com>

### CODE REVIEW

File: include/linux/virtio_shm.h [hunk 1]
```diff
@@ -10,6 +10,8 @@
 #ifdef CONFIG_VIRTIO_IVSHMEM
 struct page *virtio_shmem_allocate_page(struct device *dev);
 void virtio_shmem_free_page(struct device *dev, struct page *page);
+void *virtio_shmem_alloc(struct device *dev, size_t size);
+void virtio_shmem_free(struct device *dev, void *addr, size_t size);
```
Comment (xyzhao2018) [inline include/linux/virtio_shm.h:14]: there already has function virtio_shmem_allocate_page to alloc, why adding one more virtio_shmem_alloc?

Comment (YuanyuanZhao609) [reply include/linux/virtio_shm.h:14]: This provide virtual address, virtio_shmem_allocate_page provide page pointer for dma.

**Review by xyzhao2018 — APPROVED** [review]

**Review by xyzhao2018 — COMMENTED** [review]
Please add the tracked on JIRA ID in the commit message of each patch.

**Review by xyzhao2018 — COMMENTED** [review]
Please also add tracked-on JIRA ID and test done part.

**Comment by YuanyuanZhao609** [comment]
EB of virtio sound over ivshmem tested, works normally.

**Review by xyzhao2018 — APPROVED** [review]

**Review by xyzhao2018 — APPROVED** [review]

**Review by xyzhao2018 — APPROVED** [review]

**Review by xyzhao2018 — APPROVED** [review]

**Review by xyzhao2018 — APPROVED** [review]

**Review by xyzhao2018 — APPROVED** [review]

**Pull request merged**